lib/mind_the_gapp/mta/scheduled_arrival_time.ex

```diff
@@ -0,0 +1,45 @@
+defmodule MindTheGapp.Mta.ScheduledArrivalTime do
+  alias MindTheGapp.Repo
+  alias MindTheGapp.ScheduledArrivalTime
+  alias MindTheGapp.ScheduledArrivalTime, as: ScheduledArrivalTimeSchema
+
+  def is_data_loaded?() do
+    Repo.aggregate(ScheduledArrivalTimeSchema, :count, :id) > 0
+  end
+
+  def insert_stop_time_from_file(file_name) do
+    stop_time_file = get_stop_time_data_from(file_name)
+    persist_stop_time_data_from(stop_time_file)
+    File.close(stop_time_file)
+  end
+
+  defp persist_stop_time_data_from(stop_time_file) do
+    case stop_time_file do
+      {:ok, stop_times} ->
+        stop_times |> Enum.map(fn time -> parse_raw_stop_time(time) |> Repo.insert! end )
+      {:error, message} ->
+        IO.inspect(message)
+    end
+  end
+
+  defp get_stop_time_data_from(file_name) do
+    File.open(file_name, [:read],
+      fn file ->
+        lines = IO.read(file, :all)
+                |> String.trim_trailing("\n")
+                |> String.split("\n")
+                |> Enum.filter(fn line -> line |> String.contains?("G24") end)
+      end
```

Comment:
This is a minor thing and not something to implement, but I always try to keep resource handling (files, connections, ...) separate from the interactions with the data they have.

Reply:
I agree.